kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) Up() error {
+	if err := d.verifyFlags(); err != nil {
+		return fmt.Errorf("up could not verify flags: %s", err)
+	}
+
+	if err := d.enableComputeAPI(); err != nil {
+		return fmt.Errorf("up couldn't enable compute API: %s", err)
+	}
+
+	env := d.buildEnv()
+	script := filepath.Join(d.RepoRoot, "cluster", "kube-up.sh")
+
+	cmd := exec.Command(script)
+	cmd.SetEnv(env...)
```

Comment:
It's awesome if we can create a hermetic env for this, but are we sure we won't need any envs passed from top down.

Reply:
I'm tempted to leave it as bare-bones as possible for now, understanding that it may run into issues later (when targeting k/k rather than just cloud-provider-gcp). In tests it works for now. Totally up for either direction.